hack/proxy-infrastructure/resources.go

```diff
@@ -40,10 +52,14 @@ func vnet(conf *Config) *network.VirtualNetwork {
 		Type:     to.StringPtr("Microsoft.Network/virtualNetworks"),
 		Location: to.StringPtr(conf.region),
 	}
+	return &arm.Resource{
```

Comment:
globally, please can you change the pattern from
```
foo := thing{...}
return &arm.Resource{Resource: thing}
```
to
```
return &arm.Resource{Resource: &thing{...}}
```

notes:
* `git diff -w` will make this acceptable
* please standardise on .Resource being a pointer to a thing, not the thing itself

Reply:
Updated.

We can't do `return &arm.Resource{Resource: &thing{...}}` in 100% of cases, because there are some pieces of code where we set some values conditionally (see `vmss` functions in ARM, for example).

I mean, we can do type assertions and dereferencing (something like below), but I don't see the benifit:

```go
r := &arm.Resource{Resource: &thing{...}}

if r.Resource is a right type {
    // set values conditionally here 
}

return r
```

If you have any ideas about these cases - let me know.